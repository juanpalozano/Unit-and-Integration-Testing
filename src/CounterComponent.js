import React, { useState }from "react";
import PropTypes from "prop-types";
const CounterComponent = ({value}) => {
    const [ counter, setCounter ] = useState( 0 );
    const handdleAdd = (e) => setCounter( counter + 1 );
    const handdleSubstract = (e) => setCounter( counter - 1 );
    const handdleReset = (e) => setCounter( value );
    return (
        <div>
            <h1>CounterComponent</h1>
            <h2>{counter}</h2>
            <button onClick={ handdleAdd }>+1</button>
            <button onClick={ handdleReset }>Reset</button>
            <button onClick={ handdleSubstract }>-1</button>
        </div>
    );    
}
CounterComponent.propTypes = {
    value: PropTypes.number.isRequired
}
CounterComponent.defaultProps = {
    value: 10052021,
}
export default CounterComponent;