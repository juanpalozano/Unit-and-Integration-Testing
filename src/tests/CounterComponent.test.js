import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterComponent from '../CounterComponent';

describe('Test File: CounterComponent.js', () => {
    test('The component should display correctly the Counter Component with the default values.', () => {
        const wrapper = shallow(<CounterComponent />);
        expect(wrapper).toMatchSnapshot();
    });
    test('The component should display the value send from props.', () => {
        const counter = 100;
        const wrapper = shallow(
            <CounterComponent
                value={100}
            />
        );
        const counterDisplay = parseInt(wrapper.find('h2').text());
        expect(counterDisplay).toBe(counter);
    });
    test('The component should display the default value plus one.', () => {
        const result = 11;
        const wrapper = shallow(<CounterComponent />);
        wrapper.find('button').at(0).simulate('click');
        const counterDisplay = parseInt(wrapper.find('h2').text());
        expect(counterDisplay).toBe(result);
    });
    test('The component should display the default value minus one.', () => {
        const result = 9;
        const wrapper = shallow(<CounterComponent />);
        wrapper.find('button').at(2).simulate('click');
        const counterDisplay = parseInt(wrapper.find('h2').text());
        expect(counterDisplay).toBe(result);
    });
    test('The component should display the default value .', () => {
        const result = 100;
        const wrapper = shallow(<CounterComponent value={result}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const counterDisplay = parseInt(wrapper.find('h2').text());
        expect(counterDisplay).toBe(result);
    });
});