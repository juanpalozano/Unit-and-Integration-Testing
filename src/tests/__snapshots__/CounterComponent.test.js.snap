// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test File: CounterComponent.js The component should display correctly the Counter Component with the default values. 1`] = `
<div>
  <h1>
    CounterComponent
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</div>
`;
